test/boolean_complex_test.cpp

```diff
@@ -1030,4 +1031,159 @@ TEST(BooleanComplex, HullMask) {
+    manifold::Manifold rotated_cyl =
+        origin_cyl.Transform(manifold::RotateUp(evec));
+    manifold::Manifold right = rotated_cyl.Translate(ev1);
+    if (!right.NumTri()) continue;
+    manifold::Manifold left = c;
+    c = left.Boolean(right, manifold::OpType::Add);
+    EXPECT_TRUE(c.NumTri() > 0);
+  }
+  // Triangle Volumes
+  for (size_t i = 0; i < seeds.NumTri(); i++) {
+    int eind[3];
+    for (int j = 0; j < 3; j++) eind[j] = seeds.triVerts[i * 3 + j];
+    vec3 ev1 = vec3(seeds.vertProperties[3 * eind[0] + 0],
+                    seeds.vertProperties[3 * eind[0] + 1],
+                    seeds.vertProperties[3 * eind[0] + 2]);
+    vec3 ev2 = vec3(seeds.vertProperties[3 * eind[1] + 0],
+                    seeds.vertProperties[3 * eind[1] + 1],
+                    seeds.vertProperties[3 * eind[1] + 2]);
+    vec3 ev3 = vec3(seeds.vertProperties[3 * eind[2] + 0],
+                    seeds.vertProperties[3 * eind[2] + 1],
+                    seeds.vertProperties[3 * eind[2] + 2]);
+    vec3 a = ev1 - ev3;
+    vec3 b = ev2 - ev3;
+    vec3 n = la::normalize(la::cross(a, b));
+    // Extrude the points above and below the plane of the triangle
+    vec3 pnts[6];
+    pnts[0] = ev1 + n;
+    pnts[1] = ev2 + n;
+    pnts[2] = ev3 + n;
+    pnts[3] = ev1 - n;
+    pnts[4] = ev2 - n;
+    pnts[5] = ev3 - n;
+    // Construct the points and faces of the new manifold
+    double pts[3 * 6];
+    /* 1 */
+    pts[0] = pnts[4].x;
+    pts[1] = pnts[4].y;
+    pts[2] = pnts[4].z;
+    /* 2 */
+    pts[3] = pnts[3].x;
+    pts[4] = pnts[3].y;
+    pts[5] = pnts[3].z;
+    /* 3 */
+    pts[6] = pnts[0].x;
+    pts[7] = pnts[0].y;
+    pts[8] = pnts[0].z;
+    /* 4 */
+    pts[9] = pnts[1].x;
+    pts[10] = pnts[1].y;
+    pts[11] = pnts[1].z;
+    /* 5 */
+    pts[12] = pnts[5].x;
+    pts[13] = pnts[5].y;
+    pts[14] = pnts[5].z;
+    /* 6 */
+    pts[15] = pnts[2].x;
+    pts[16] = pnts[2].y;
+    pts[17] = pnts[2].z;
+    int faces[24];
+    // 1 2 5
+    faces[0] = 0;
+    faces[1] = 1;
+    faces[2] = 4;
+    // 3 4 6
+    faces[3] = 2;
+    faces[4] = 3;
+    faces[5] = 5;
+    // 2 1 4
+    faces[6] = 1;
+    faces[7] = 0;
+    faces[8] = 3;
+    // 4 3 2
+    faces[9] = 3;
+    faces[10] = 2;
+    faces[11] = 1;
+    // 4 1 5
+    faces[12] = 3;
+    faces[13] = 0;
+    faces[14] = 4;
+    // 5 6 4
+    faces[15] = 4;
+    faces[16] = 5;
+    faces[17] = 3;
+    // 6 5 2
+    faces[18] = 5;
+    faces[19] = 4;
+    faces[20] = 1;
+    // 2 3 6
+    faces[21] = 1;
+    faces[22] = 2;
+    faces[23] = 5;
+    manifold::MeshGL64 tri_m;
+    for (int j = 0; j < 18; j++)
+      tri_m.vertProperties.insert(tri_m.vertProperties.end(), pts[j]);
+    for (int j = 0; j < 24; j++)
+      tri_m.triVerts.insert(tri_m.triVerts.end(), faces[j]);
+    manifold::Manifold left = c;
+    manifold::Manifold right(tri_m);
+    if (!right.NumTri()) continue;
+    c = left.Boolean(right, manifold::OpType::Add);
+    EXPECT_TRUE(c.NumTri() > 0);
```

Comment:
If this is affecting the timing, let's move it out of the loop so that our CSG tree can optimize its operations. No need for this test to pinpoint the location of the problem - it just needs to tell if anything went wrong. Also, a `Status` check is probably a more canonical way to handle this.

Reply:
Hmm.  Without the NumTri() calls, things complete very quickly - it might not be slow enough to usefully flag the problem?  (Assuming the problem is the larger triangle counts due to the tolerance issues...)

It's more than possible that my BRL-CAD side implementation of all this is sub-optimal (i.e. doing things I shouldn't during the iterations on the boolean processing), so I suppose another thing for me to try is to "back port" how this test is doing things into our code and see if that fixes the speed issues.

Reply:
OK, looks like Status() is slow as well (at least here) so it's not NumTri() specifically.

Reply:
yeah, @pca006132 has done some pretty awesome optimizations of our CSG tree that you certainly don't want to skip just for a debug check. Would you mind moving that check out of the loop so I can see the CI's timing? It'd be great if that's the whole problem, but I'd bet it's not.

Reply:
Yeah, every check (num tri, status, genus, etc) requires the manifold to be evaluated -it's the forced evaluation before we can reorder them that causes the slowness.